File: Blanks/OptionsViewController.m
//
//  FlipsideViewController.m
//  Blanks
//
//  Created by Kai Kunze on 19/10/2013.
//  Copyright (c) 2013 Kai Kunze. All rights reserved.
//

#import "OptionsViewController.h"
#import "MoreBlanks-Swift.h"

@interface OptionsViewController ()
@property (weak, nonatomic) IBOutlet UISwitch *dragSwitch;

@end

@implementation OptionsViewController


- (IBAction)buyCoffee:(id)sender {
    UIViewController *vc = [SwiftUIViewFactory makeSwiftUIViewWithDismissHandler:^{
        [[self presentedViewController] dismissViewControllerAnimated:YES completion:nil];
    }];
    [self presentViewController:vc animated:YES completion:nil];
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    NSUserDefaults* defaults = [NSUserDefaults standardUserDefaults];
    bool tapping =  self.dragSwitch.on;
    
    if ([defaults objectForKey:@"TappingUI"] != nil) {
        tapping = [defaults boolForKey:@"TappingUI"];
    }
        
    self.dragSwitch.on  = tapping;
	// Do any additional setup after loading the view, typically from a nib.
}

- (void) viewDidAppear:(BOOL)animated{

}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark - Actions
- (IBAction)toggleDragging:(id)sender {
    NSUserDefaults *prefs = [NSUserDefaults standardUserDefaults];
    //NSLog(@"%d",self.dragSwitch.isOn);
    [prefs setBool:self.dragSwitch.isOn forKey:@"TappingUI"];
    [prefs synchronize];
    //NSLog(@"%d",[prefs boolForKey:@"theBoolKey"]);
}

- (IBAction)done:(id)sender
{
    [self.delegate optionsViewControllerDidFinish:self];
}

@end

File: Blanks/OptionsViewController.swift
//
//  OptionsViewController.swift
//  Blanks
//
//  Created by Kai Kunze on 19/10/2013.
//  Copyright (c) 2013 Kai Kunze. All rights reserved.
//

import UIKit

protocol OptionsViewControllerDelegate: AnyObject {
    func optionsViewControllerDidFinish(_ controller: OptionsViewController)
}

final class OptionsViewController: UIViewController {
    @IBOutlet private weak var dragSwitch: UISwitch!
    weak var delegate: OptionsViewControllerDelegate?

    override func viewDidLoad() {
        super.viewDidLoad()

        let defaults = UserDefaults.standard
        let tapping = defaults.object(forKey: .tappingUI) != nil
            ? defaults.bool(forKey: .tappingUI)
            : false

        dragSwitch.isOn = tapping
    }

    // MARK: - Actions

    @IBAction func toggleDragging(_ sender: Any) {
        let prefs = UserDefaults.standard
        prefs.set(dragSwitch.isOn, forKey: .tappingUI)
    }

    @IBAction func done(_ sender: Any) {
        delegate?.optionsViewControllerDidFinish(self)
    }
}
